django/cantusdb_project/main_app/tests/test_views.py

```diff
@@ -0,0 +1,156 @@
+        response = self.client.get(reverse("indexer-list"))
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_correct_templates(self):
+        response = self.client.get(reverse("indexer-list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "base.html")
+        self.assertTemplateUsed(response, "indexer_list.html")
+
+    def test_pagination(self):
+        # To get total number of pages do a ceiling integer division
+        q, r = divmod(self.number_of_indexers, self.PAGE_SIZE)
+        pages = q + bool(r)
+
+        # Test all pages
+        for page_num in range(1, pages + 1):
+            response = self.client.get(reverse("indexer-list"), {"page": page_num})
+            self.assertEqual(response.status_code, 200)
+            self.assertTrue("is_paginated" in response.context)
+            self.assertTrue(response.context["is_paginated"] == True)
+            if page_num == pages and (self.number_of_indexers % self.PAGE_SIZE != 0):
+                self.assertTrue(
+                    len(response.context["indexers"])
+                    == (self.number_of_indexers % self.PAGE_SIZE)
+                )
+            else:
+                self.assertTrue(len(response.context["indexers"]) == self.PAGE_SIZE)
+
+        # Test the "last" syntax
+        response = self.client.get(reverse("indexer-list"), {"page": "last"})
+        self.assertEqual(response.status_code, 200)
+
+        # Test some invalid values for pages
+        response = self.client.get(reverse("indexer-list"), {"page": -1})
+        self.assertEquals(response.status_code, 404)
+        response = self.client.get(reverse("indexer-list"), {"page": 0})
+        self.assertEquals(response.status_code, 404)
+        response = self.client.get(reverse("indexer-list"), {"page": "lst"})
+        self.assertEquals(response.status_code, 404)
+        response = self.client.get(reverse("indexer-list"), {"page": pages + 1})
+        self.assertEquals(response.status_code, 404)
+
+    def test_search(self):
+        number_of_indexers = Indexer.objects.count()
+
+        # Search by first name
+        random_indexer = Indexer.objects.get(
+            id=random.randint(1, number_of_indexers + 1)
+        )
+
+        # Search the whole first name
+        response = self.client.get(
+            reverse("indexer-list"), {"q": random_indexer.first_name}
+        )
+        self.assertEquals(response.status_code, 200)
+        # Check object_list (which has the whole queryset, not paginated) instead of
+        # indexers which is paginated and might not contain random_indexer if it
+        # is not on the first page
+        self.assertTrue(random_indexer in response.context["paginator"].object_list)
+
+        # Search for a three letter slice of the first name
+        first_name = random_indexer.first_name
+        first_name_length = len(first_name)
+        if first_name_length <= 3:
```

Comment:
Why not just test `if first_name_length > 3:` and remove the pass?

Reply:
Good point, I'll rewrite it